sound/soc/sof/ipc.c

```diff
@@ -696,14 +697,27 @@ int snd_sof_ipc_set_get_comp_data(struct snd_sof_ipc *ipc,
 
 	/* send normal size ipc in one part */
 	if (cdata->rhdr.hdr.size <= SOF_IPC_MSG_MAX_SIZE) {
+		/* Allocate temporary buffer for new cdata */
+		reply_cdata = kzalloc(cdata->rhdr.hdr.size, GFP_KERNEL);
+		if (!reply_cdata)
+			return -ENOMEM;
+
+		/* Copy header to reply */
+		memcpy(reply_cdata, cdata, sizeof(*cdata));
+
 		err = sof_ipc_tx_message(sdev->ipc, cdata->rhdr.hdr.cmd, cdata,
-					 cdata->rhdr.hdr.size, cdata,
-					 cdata->rhdr.hdr.size);
+					 cdata->rhdr.hdr.size, reply_cdata,
+					 reply_cdata->rhdr.hdr.size);
 
 		if (err < 0)
 			dev_err(sdev->dev, "error: set/get ctrl ipc comp %d\n",
 				cdata->comp_id);
+		else
+			if (!send)
```

Comment:
Please, put "else" and "if" on the same line, you'll save one level of indentation that way.

Reply:
Done, good to know this. Line lengths are a pain so good to get some help from coding style.